llms/mlx_lm/utils.py

```diff
@@ -679,13 +690,31 @@ def quantize_model(
         config (dict): Model configuration.
         q_group_size (int): Group size for quantization.
         q_bits (int): Bits per weight for quantization.
+        quant_predicate (Callable): A callable that decides how
+            to quantize each layer based on the path.
+            Accepts the layer `path`, the `module` and the model `config`.
+            Returns either a bool to signify quantize/no quantize or
+            a dict of quantization parameters to pass to `to_quantized`.
 
     Returns:
         Tuple: Tuple containing quantized weights and config.
     """
     quantized_config = copy.deepcopy(config)
-    nn.quantize(model, q_group_size, q_bits)
     quantized_config["quantization"] = {"group_size": q_group_size, "bits": q_bits}
+
+    # Add any custom quantization parameters to the config as we go
+    def _class_predicate(p, m):
+        bool_or_params = quant_predicate(p, m, config)
+        if isinstance(bool_or_params, dict):
+            quantized_config["quantization"][p] = bool_or_params
+        return bool_or_params
```

Comment:
Should we be storing the `True` / `False` values here as well? If we don't store them, then we can't  reconstruct quants that use `True` / `False` but the default quantization parameters from the config right?

Reply:
Great point! Just fixed it.

Reply:
Sorry, I may have mislead you here.  This check:

> return f"{p}.scales" in weights

should take care of the weights which are not quantized when we load. Up to you if you want to keep the explicit config for those. I can see how it can be useful.